dll/win32/shell32/CDefView.cpp

```diff
@@ -3069,11 +3120,190 @@ HRESULT WINAPI CDefView::AddPropertySheetPages(DWORD dwReserved, LPFNADDPROPSHEE
     return S_OK;
 }
 
+static HRESULT Read(IStream *pS, LPVOID buffer, ULONG cb)
+{
+    ULONG read;
+    HRESULT hr = pS->Read(buffer, cb, &read);
+    return FAILED(hr) ? hr : (cb == read ? S_OK : HResultFromWin32(ERROR_MORE_DATA));
```

Comment:
Why do you need this extra HRESULT conversion?
What about [`IStream_Read() aka. SHIStream_Read()`](https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/shlwapi/istream.c;hb=957bb89548ba90484b67c553559b92405eb8e39c#l541) ?

Reply:
I can't call the helper function in shlwapi for whatever reason. Probably related to wine code being C. The correct name to use outside shlwapi would be IStream_Read but wine calls it SHIStream_Read because C has those COM helper defines, one of them being IStream_Read.

Reply:
> The correct name to use outside shlwapi would be IStream_Read [...]

we are outside of shlwapi here, so maybe it would work? (since that IStream_Read function is exported).... unless it would use here the autogenerated macros from the headers? Dunno.